src/main/java/org/quantumbadger/redreader/activities/MainActivity.java

```diff
@@ -950,13 +1111,23 @@ public boolean onOptionsItemSelected(final MenuItem item) {
 
 		switch(item.getItemId()) {
 			case android.R.id.home:
-				onBackPressed();
+				onHomePressed();
 				return true;
 			default:
 				return super.onOptionsItemSelected(item);
 		}
 	}
 
+	private void onHomePressed() {
+		if(PrefsUtility.appearance_navigation_type(this, sharedPreferences).equals("drawer_tabs")) {
+			if(mDrawer.isDrawerOpen(GravityCompat.START))
+				mDrawer.closeDrawers();
+			else
+				mDrawer.openDrawer(GravityCompat.START);
+		} else
+			onBackPressed();
```

Comment:
Minor point, but it would be good to use curly braces around all statement blocks like this, for clarity.